### PR TITLE
Examples: Added OimoPhysics instancing example. Updated OimoPhysics

### PR DESCRIPTION
**Pull request by VBT-YTokan**
Related to pull request #21679, which was closed because of LGTM issues in OimoPhysics code. Here is the updated one and cleaned up manually to have 0 issues.

- cleaned up OimoPhysics code;
- added physics wrapper;

This contribution is funded by [VeryBigThings](https://verybigthings.com).

image

### CODE REVIEW

File: examples/physics_oimo_instancing.html [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+		stats.update();
+
+	}
+
+</script>
+</body>
+</html>
+
+<!--<!DOCTYPE html>-->
```
Comment (Mugen87) [inline examples/physics_oimo_instancing.html:158]: What is the purpose of this commented out code?

Normally, such code sections are not added to the examples.

**Comment by Mugen87** [comment]
https://raw.githack.com/VeryBigThings/three.js/examples/physics_oimo_instancing/examples/physics_oimo_instancing.html

File: examples/physics_oimo_instancing.html (hunk 1 above)
Comment (VBT-YTokan) [reply examples/physics_oimo_instancing.html]: Removed the commented out code.
Thank you

**Comment by Mugen87** [comment]
When testing the example it seems that when you refresh, the objects bounce sometimes in a strange way from the ground or each other right at the beginning of the simulation. 

It did no happen very often (maybe two out of ten refreshes) but it looks unnatural and I don't see comparable glitches in the [ammo demo](https://threejs.org/examples/physics_ammo_instancing).

**Comment by OleksiiMaksymovVBT** [comment]
@Mugen87 it seems, like the issue is happening, when `world.delta(timeDelta)` param is non-constant, normally we work with 1/60 value, so if there is some drop by user, physics will just slow done. What do you think if change it like this here?

**Comment by Mugen87** [comment]
Not sure I understand. The ammo demo assumes a fix framerate:

https://github.com/mrdoob/three.js/blob/09a68603cda2e17bf1a60988fa124132ee5f86fa/examples/jsm/physics/AmmoPhysics.js#L12

And `OimoPhysics.js` does the same. So it appears to me the error lies elsewhere?

**Comment by OleksiiMaksymovVBT** [comment]
@Mugen87 nope, it has next line 
https://github.com/mrdoob/three.js/blob/09a68603cda2e17bf1a60988fa124132ee5f86fa/examples/jsm/physics/AmmoPhysics.js#L189
@mrdoob also added such line to OimoPhysics previously, so it sets only setInterval timeout, but if there is some framedrop, that stuck setInterval pipeline -- it can be larger, than 60 fps

**Comment by Mugen87** [comment]
But I see the same line here:

https://github.com/mrdoob/three.js/pull/21851/files#diff-3912adf4d9c30a30ea11d2f86070013e9c091818e766140d55239fcb823c6181R147

**Comment by OleksiiMaksymovVBT** [comment]
@Mugen87 exactly, that line was written firstly by @mrdoob. 
We are gonna push the new version with constant value, 10 mins :)

**Comment by OleksiiMaksymovVBT** [comment]
Yeap, here is new version available, @Mugen87 
https://raw.githack.com/VeryBigThings/three.js/examples/physics_oimo_instancing/examples/physics_oimo_instancing.html

File: examples/jsm/physics/OimoPhysics.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+	let lastTime = 0;
+
+	function step() {
+
+		const time = performance.now();
+
+		if ( lastTime > 0 ) {
+
+			const delta = ( time - lastTime ) / 1000;
```
Comment (Mugen87) [inline examples/jsm/physics/OimoPhysics.js:147]: This line can now be removed, right? Same for the variables `lastTime`  and `time`.

**Comment by Mugen87** [comment]
It seems the physics glitches are now gone 🎉 .

**Comment by OleksiiMaksymovVBT** [comment]
@Mugen87 yeap, that is was I talking about. Are we ready to add here more examples and also physics debugger? 🙂

**Comment by Mugen87** [comment]
Well, I would say one example is sufficient for now. Let's see what @mrdoob thinks about this.

**Comment by mrdoob** [comment]
I think it'll be good to have more examples. Try to keep them simple if you can though.

**Comment by mrdoob** [comment]
Thanks!

**Comment by OleksiiMaksymovVBT** [comment]
@mrdoob let's just not revert it again 😅😂

**Pull request merged**